routes/bridges/gassistant/index.js

```diff
@@ -0,0 +1,118 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of ThingEngine
+//
+// Copyright 2017-2019 The Board of Trustees of the Leland Stanford Junior University
+//
+// Author: Swee Kiat Lim <sweekiat@cs.stanford.edu>
+//
+// See COPYING for details
+"use strict";
+
+const Q = require('q');
+const express = require('express');
+
+const EngineManager = require('../../../almond/enginemanagerclient');
+const { actionssdk, Image } = require('actions-on-google');
+
+const userUtils = require('../../../util/user');
+
+var router = express.Router();
+
+class GoogleAssistantDelegate {
+    constructor(locale) {
+        this._buffer = '';
+        this._locale = locale;
+    }
+
+    send(text, icon) {
+        this._buffer += text + '\n';
+    }
+
+    sendPicture(url, icon) {
+        this._image = new Image({
+            url: url,
+            alt: url,
+        });
+    }
+
+    sendRDL(rdl, icon) {
+        this._buffer += rdl.displayTitle + '\n';
+    }
+
+    sendChoice(idx, what, title, text) {
+        this._buffer += title + '\n';
+    }
+
+    sendButton(title, json) {
+        // FIXME
+    }
+
+    sendLink(title, url) {
+        if (url === '/user/register') {
+            this._card = {
+                type: 'LinkAccount'
+            };
+        }
+        // FIXME handle other URL types
+    }
+
+    sendResult(message, icon) {
+        this._buffer += message.toLocaleString(this._locale) + '\n';
+    }
+
+    sendAskSpecial(what) {
+        this._askSpecial = what;
+    }
+}
+GoogleAssistantDelegate.prototype.$rpcMethods = ['send', 'sendPicture', 'sendChoice', 'sendLink', 'sendButton', 'sendAskSpecial', 'sendRDL', 'sendResult'];
+
+const app = actionssdk();
+
+// Register handler for Actions SDK
+
+app.intent('actions.intent.MAIN', (conv) => {
+    conv.ask('Hello! I\'m Almond, your virtual assistant.');
+});
+
+app.intent('actions.intent.TEXT', (conv, input) => {
+    if (input === 'bye' || input === 'goodbye')
+        return conv.close('See you later!');
+
+    let anonymous = true;
+    // const userId = conv.user._id || 'UserID';
+    const locale = conv.body.user.locale;
+    const conversationId = conv.body.conversation.conversationId;
```

Comment:
What's the format of conv.body? Can we have a comment? Or even better, a test with a few example bodies, like the Alexa tests?